src/main/scala/codacy/docker/api/utils/ToolHelper.scala

```diff
@@ -0,0 +1,36 @@
+package codacy.docker.api.utils
+
+import codacy.docker.api._
+
+object ToolHelper {
+
+  def filesToLint(source: Source.Directory, filesOpt: Option[Set[Source.File]]): Set[String] = {
+    filesOpt.fold(Set(source.path)) { files =>
+      files.map(_.path)
+    }
+  }
+
+  def patternsToLint(conf: Option[List[Pattern.Definition]])(implicit specification: Tool.Specification): Option[List[Pattern.Definition]] = {
```

Comment:
Quick note, when the tool runs with a config file, our conf will be empty, so this will be None, and each docker handles the None